### PR TITLE
GLO: Added task for data compression monitoring for all detectors

### PR DESCRIPTION
**Pull request by tklemenz**
This PR adds a qc task for monitoring the entropy compression and compression factors of the detectors. It can be used for any number of detectors at a time.

In addition, I added Utils to Common with a function for getting parameters from taskParameters in the config easily. The function was recently introduced in the TPC code but I think it should rather be in a common place. I left it also in TPC/Utility for now for compatibility reasons. Later this will be removed and the corresponding tasks using the function will be adjusted.

### CODE REVIEW

File: Modules/GLO/src/DataCompressionQcTask.cxx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+#include "Framework/ProcessingContext.h"
+#include <Framework/InputRecord.h>
+
+// QC includes
+#include "QualityControl/QcInfoLogger.h"
+#include "GLO/DataCompressionQcTask.h"
+#include "Common/Utils.h"
+
+// root includes
+#include "TH1.h"
```
Comment (wiechula) [inline Modules/GLO/src/DataCompressionQcTask.cxx:27]: Doesn't `unique_ptr` need the header? So perhaps move this to the header.

File: Modules/GLO/src/DataCompressionQcTask.cxx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  TH1::AddDirectory(false);
+
+  const auto nBins = o2::quality_control_modules::common::getFromConfig<int>(mCustomParameters, "nBins");
+  const auto xMin = o2::quality_control_modules::common::getFromConfig<int>(mCustomParameters, "xMin");
+  const auto xMax = o2::quality_control_modules::common::getFromConfig<int>(mCustomParameters, "xMax");
+
+  // initialize histograms for active detectors
+  bool useAll = o2::quality_control_modules::common::getFromConfig<bool>(mCustomParameters, "useAll");
+  for (const auto& det : detList) {
+    bool useDet = o2::quality_control_modules::common::getFromConfig<bool>(mCustomParameters, det);
```
Comment (wiechula) [inline Modules/GLO/src/DataCompressionQcTask.cxx:48]:
```suggestion
for (int iDet = 0; iDet < o2::detectors::DetID::nDetectors; ++iDet) {
    const bool useDet = o2::quality_control_modules::common::getFromConfig<bool>(mCustomParameters, o2::detectors::DetID::getName(iDet));
```

File: Modules/GLO/src/DataCompressionQcTask.cxx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    for (const auto& hist : pair.second) {
+      hist->Reset();
+    }
+  }
+}
+
+void DataCompressionQcTask::processMessage(const o2::ctf::CTFIOSize& ctfEncRep, const std::string detector)
+{
+  auto entropyCompression = (ctfEncRep.ctfIn - ctfEncRep.ctfOut) / float(ctfEncRep.ctfIn);
+  auto compression = (ctfEncRep.rawIn - ctfEncRep.ctfOut) / float(ctfEncRep.rawIn);
```
Comment (wiechula) [inline Modules/GLO/src/DataCompressionQcTask.cxx:136]:
```suggestion
  const auto entropyCompression = float(ctfEncRep.ctfIn - ctfEncRep.ctfOut) / float(ctfEncRep.ctfIn);
  const auto compression = float(ctfEncRep.rawIn - ctfEncRep.ctfOut) / float(ctfEncRep.rawIn);
```

**Review by wiechula — CHANGES_REQUESTED** [review]

File: Modules/GLO/src/DataCompressionQcTask.cxx (hunk 1 above)
Comment (tklemenz) [reply Modules/GLO/src/DataCompressionQcTask.cxx]: It actually worked with the forward declaration in the case of `TH1`. For `TCanvas` the include was needed in the header. I moved the TH1 include to the header now anyway.

**Comment by tklemenz** [comment]
@knopers8 @wiechula @Barthelemy The error looks unrelated to me.

**Pull request merged**